test/SymbolGraph/Relationships/Synthesized/RemoteInheritedDocs.swift

```diff
@@ -53,9 +48,6 @@ import RemoteP
 // but `otherFunc` does. Regardless, each one needs a `sourceOrigin` field connecting it to
 // upstream.
 
-// `RemoteP.P` also has an extension with a default implementation for `extraFunc` that does have
-// docs, but overriding it here should prevent those from appearing
-
```

Comment:
@QuietMisdreavus This comments seems to imply the exact opposite of what I want to achieve in this PR. Is there any reason why we didn’t want to inherit documentation from extensions?

For context, the concrete example I want to fix with this PR is that this implementation of `_ArrayProtocol.filter` (https://github.com/apple/swift/blob/c531f2914fda0b84d8af091a0081799101f76d9b/stdlib/public/core/ArrayType.swift#L73-L78) should inherit the documentation from the documentation of `Sequence.filter`, which is implemented in an extension (https://github.com/apple/swift/blob/5df0f053d95c1143ac34ac88faafdfa3902a3224/stdlib/public/core/Sequence.swift#L717-L739)

Reply:
cc @franklinsch (and possibly also @bitjammer for the underlying discussion of inheriting docs in the first place)

I think it should respect the presence of the `-skip-inherited-docs` flag, but otherwise i don't remember off the top of my head.

Reply:
I think we already respect the `-skip-inherited-docs` flag because we’re not calling into `getDocCommentProvidingDecl` if it is set and this PR only modifies `getDocCommentProvidingDecl`

https://github.com/apple/swift/blob/9fdb0e79e2dab72fdc01f8fa50d9cc249a48a5a0/lib/SymbolGraphGen/Symbol.cpp#L359-L365